### DISCUSSION TITLE
COSMO Second Generation

### DISCUSSION THREAD

**Discussion by effeminati** [gmtsar/gmtsar#885] [Q&A]
COSMO Second Generation with COSMO-SkyMed are nowdays the main constellation for high resolution interferometric X-band data. I have not found any possibility to use COSMO Second Generation data (also in conjuntion with COSMO-SkyMed) in GMTSAR. 
Therefore I think it could be very important for many users an upgrade of GMTSAR to perform this processing. 
Is there any planned update to enable the use of COSMO Second Generation data in GMTSAR? In case when?

**Comment by Xiaohua-Eric-Xu** [comment]
I had a very long conversation with a guy from Turkey, and worked out a solution for CSG data. I think the current CSK-SLC option works with the CSG data except you may need to tune the coregistration parameters in p2p_processing.csh. I think the attached script works, and there's a SLC_factor one needs to experiment with in order to fully utilize the dynamic range of short integers (SLC format in GMTSAR). Feel free to chat more or send me any problems you get processing CSG data. 

[p2p_processing_CSK.csh.txt](https://github.com/gmtsar/gmtsar/files/14662918/p2p_processing_CSK.csh.txt)
